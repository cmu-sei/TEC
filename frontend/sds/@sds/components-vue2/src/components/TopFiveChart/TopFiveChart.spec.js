// TEC - Automated Mismatch Detection Tool
// Copyright 2023 Carnegie Mellon University.
// NO WARRANTY. THIS CARNEGIE MELLON UNIVERSITY AND SOFTWARE ENGINEERING INSTITUTE MATERIAL IS FURNISHED ON AN "AS-IS" BASIS. CARNEGIE MELLON UNIVERSITY MAKES NO WARRANTIES OF ANY KIND, EITHER EXPRESSED OR IMPLIED, AS TO ANY MATTER INCLUDING, BUT NOT LIMITED TO, WARRANTY OF FITNESS FOR PURPOSE OR MERCHANTABILITY, EXCLUSIVITY, OR RESULTS OBTAINED FROM USE OF THE MATERIAL. CARNEGIE MELLON UNIVERSITY DOES NOT MAKE ANY WARRANTY OF ANY KIND WITH RESPECT TO FREEDOM FROM PATENT, TRADEMARK, OR COPYRIGHT INFRINGEMENT.
// Released under a BSD (SEI)-style license, please see license.txt or contact permission@sei.cmu.edu for full terms.
// [DISTRIBUTION STATEMENT A] This material has been approved for public release and unlimited distribution.  Please see Copyright notice for non-US Government use and distribution.
// This Software was created using the SEI Design System and includes and/or makes use of Third-Party Software subject to its own license.
// DM23-003
 
 
import { shallowMount } from '@vue/test-utils'
import Component from "./TopFiveChart.vue";

describe("TopFiveChart.vue", () => {
  it("is a Vue instance", () => {
    const wrapper = shallowMount(Component);
    expect(wrapper.vm).toBeTruthy();
  });

  it("matches snapshot with no props assigned", () => {
    const propsData = {};
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with urls", () => {
    const propsData = {
      title: "Top Five Chart (with urls, view all url, showPercent)",
      viewAllUrl: "https://seinet.sei.cmu.edu",
      entries: [
        {
          id: 1,
          title: "Test 1",
          count: 40,
          url: "https://seinet.sei.cmu.edu",
        },
        {
          id: 2,
          title: "Test 2",
          count: 30,
          url: "https://seinet.sei.cmu.edu",
        },
        {
          id: 3,
          title: "Test 3",
          count: 15,
          url: "https://seinet.sei.cmu.edu",
        },
        {
          id: 4,
          title: "Test 4",
          count: 10,
          url: "https://seinet.sei.cmu.edu",
        },
        { id: 5, title: "Test 5", count: 3, url: "https://seinet.sei.cmu.edu" },
        { id: 6, title: "Test 6", count: 2, url: "https://seinet.sei.cmu.edu" },
      ],
      showPercent: true,
      progressColor: "cyan",
    };
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with urls, doNotLinkEntries", () => {
    const propsData = {
      title: "Top Five Chart (with urls, doNotLinkEntries)",
      entries: [
        {
          id: 1,
          title: "Test 1",
          count: 40,
          url: "https://seinet.sei.cmu.edu",
        },
        {
          id: 2,
          title: "Test 2",
          count: 30,
          url: "https://seinet.sei.cmu.edu",
        },
        {
          id: 3,
          title: "Test 3",
          count: 15,
          url: "https://seinet.sei.cmu.edu",
        },
        {
          id: 4,
          title: "Test 4",
          count: 10,
          url: "https://seinet.sei.cmu.edu",
        },
        { id: 5, title: "Test 5", count: 3, url: "https://seinet.sei.cmu.edu" },
        { id: 6, title: "Test 6", count: 2, url: "https://seinet.sei.cmu.edu" },
      ],
      doNotLinkEntries: true,
    };
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with no urls", () => {
    const propsData = {
      title: "Top Five Chart (no urls)",
      entries: [
        { id: 1, title: "Test 1", count: 40 },
        { id: 2, title: "Test 2", count: 30 },
      ],
      progressColor: "orange",
    };
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with no data", () => {
    const propsData = {
      title: "Top Five Chart (no data)",
      entries: [],
    };
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with malformed data", () => {
    const propsData = {
      title: "Top Five Chart (malformed data)",
      entries: [
        { id: 1, title: "Test 1", count: 40 },
        { id: 2, title: "Test 2" },
      ],
    };
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
