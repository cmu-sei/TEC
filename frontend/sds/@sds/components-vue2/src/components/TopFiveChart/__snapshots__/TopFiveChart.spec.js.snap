// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopFiveChart.vue matches snapshot with malformed data 1`] = `
"<div data-id=\\"sds-top-five-chart\\" class=\\"space-y-4 chart\\">
  <h3 class=\\"text-2xl font-bold\\">
    Top Five Chart (malformed data)
  </h3>
  <div>
    <p>
      The chart data is malformed and cannot be displayed at this time.
    </p>
  </div>
</div>"
`;

exports[`TopFiveChart.vue matches snapshot with no data 1`] = `
"<div data-id=\\"sds-top-five-chart\\" class=\\"space-y-4 chart\\">
  <h3 class=\\"text-2xl font-bold\\">
    Top Five Chart (no data)
  </h3>
  <div>
    <p>
      There is no chart data to display at this time.
    </p>
  </div>
</div>"
`;

exports[`TopFiveChart.vue matches snapshot with no props assigned 1`] = `
"<div data-id=\\"sds-top-five-chart\\" class=\\"space-y-4 chart\\">
  <h3 class=\\"text-2xl font-bold\\">

  </h3>
  <div>
    <p>
      There is no chart data to display at this time.
    </p>
  </div>
</div>"
`;

exports[`TopFiveChart.vue matches snapshot with no urls 1`] = `
"<div data-id=\\"sds-top-five-chart\\" class=\\"space-y-4 chart\\">
  <h3 class=\\"text-2xl font-bold\\">
    Top Five Chart (no urls)
  </h3>
  <div>
    <ul class=\\"space-y-6\\">
      <li>
        <div class=\\"flex\\">
          <div class=\\"flex-grow\\">
            <div class=\\"mb-2\\">
              <div role=\\"progressbar\\" title=\\"40\\" aria-valuenow=\\"40\\" aria-valuemin=\\"0\\" aria-valuemax=\\"40\\" class=\\"h-6 mr-2 rounded bg-orange-900\\" style=\\"width: 100%;\\"><span class=\\"sr-only\\">40
                  Test 1</span></div>
            </div>
            <div class=\\"text-sm font-semibold chart-label-section\\">
              40
              ·
              <a href=\\"#\\" class=\\"hover:underline focus:underline focus:outline-none\\">Test 1</a></div>
          </div>
        </div>
      </li>
      <li>
        <div class=\\"flex\\">
          <div class=\\"flex-grow\\">
            <div class=\\"mb-2\\">
              <div role=\\"progressbar\\" title=\\"30\\" aria-valuenow=\\"30\\" aria-valuemin=\\"0\\" aria-valuemax=\\"40\\" class=\\"h-6 mr-2 rounded bg-orange-700\\" style=\\"width: 75%;\\"><span class=\\"sr-only\\">30
                  Test 2</span></div>
            </div>
            <div class=\\"text-sm font-semibold chart-label-section\\">
              30
              ·
              <a href=\\"#\\" class=\\"hover:underline focus:underline focus:outline-none\\">Test 2</a></div>
          </div>
        </div>
      </li>
      <!---->
    </ul>
  </div>
</div>"
`;

exports[`TopFiveChart.vue matches snapshot with urls 1`] = `
"<div data-id=\\"sds-top-five-chart\\" class=\\"space-y-4 chart\\">
  <h3 class=\\"text-2xl font-bold\\">
    Top Five Chart (with urls, view all url, showPercent)
  </h3>
  <div>
    <ul class=\\"space-y-6\\">
      <li>
        <div class=\\"flex\\">
          <div class=\\"flex-grow\\">
            <div class=\\"mb-2\\">
              <div role=\\"progressbar\\" title=\\"40\\" aria-valuenow=\\"40\\" aria-valuemin=\\"0\\" aria-valuemax=\\"40\\" class=\\"h-6 mr-2 rounded bg-blue-900\\" style=\\"width: 100%;\\"><span class=\\"sr-only\\">40%
                  Test 1</span></div>
            </div>
            <div class=\\"text-sm font-semibold chart-label-section\\">
              40%
              ·
              <a href=\\"https://seinet.sei.cmu.edu\\" class=\\"hover:underline focus:underline focus:outline-none\\">Test 1</a></div>
          </div>
        </div>
      </li>
      <li>
        <div class=\\"flex\\">
          <div class=\\"flex-grow\\">
            <div class=\\"mb-2\\">
              <div role=\\"progressbar\\" title=\\"30\\" aria-valuenow=\\"30\\" aria-valuemin=\\"0\\" aria-valuemax=\\"40\\" class=\\"h-6 mr-2 rounded bg-blue-700\\" style=\\"width: 75%;\\"><span class=\\"sr-only\\">30%
                  Test 2</span></div>
            </div>
            <div class=\\"text-sm font-semibold chart-label-section\\">
              30%
              ·
              <a href=\\"https://seinet.sei.cmu.edu\\" class=\\"hover:underline focus:underline focus:outline-none\\">Test 2</a></div>
          </div>
        </div>
      </li>
      <li>
        <div class=\\"flex\\">
          <div class=\\"flex-grow\\">
            <div class=\\"mb-2\\">
              <div role=\\"progressbar\\" title=\\"15\\" aria-valuenow=\\"15\\" aria-valuemin=\\"0\\" aria-valuemax=\\"40\\" class=\\"h-6 mr-2 rounded bg-blue-500\\" style=\\"width: 37.5%;\\"><span class=\\"sr-only\\">15%
                  Test 3</span></div>
            </div>
            <div class=\\"text-sm font-semibold chart-label-section\\">
              15%
              ·
              <a href=\\"https://seinet.sei.cmu.edu\\" class=\\"hover:underline focus:underline focus:outline-none\\">Test 3</a></div>
          </div>
        </div>
      </li>
      <li>
        <div class=\\"flex\\">
          <div class=\\"flex-grow\\">
            <div class=\\"mb-2\\">
              <div role=\\"progressbar\\" title=\\"10\\" aria-valuenow=\\"10\\" aria-valuemin=\\"0\\" aria-valuemax=\\"40\\" class=\\"h-6 mr-2 rounded bg-blue-300\\" style=\\"width: 25%;\\"><span class=\\"sr-only\\">10%
                  Test 4</span></div>
            </div>
            <div class=\\"text-sm font-semibold chart-label-section\\">
              10%
              ·
              <a href=\\"https://seinet.sei.cmu.edu\\" class=\\"hover:underline focus:underline focus:outline-none\\">Test 4</a></div>
          </div>
        </div>
      </li>
      <li>
        <div class=\\"flex\\">
          <div class=\\"flex-grow\\">
            <div class=\\"mb-2\\">
              <div role=\\"progressbar\\" title=\\"3\\" aria-valuenow=\\"3\\" aria-valuemin=\\"0\\" aria-valuemax=\\"40\\" class=\\"h-6 mr-2 rounded bg-blue-100\\" style=\\"width: 7.5%;\\"><span class=\\"sr-only\\">3%
                  Test 5</span></div>
            </div>
            <div class=\\"text-sm font-semibold chart-label-section\\">
              3%
              ·
              <a href=\\"https://seinet.sei.cmu.edu\\" class=\\"hover:underline focus:underline focus:outline-none\\">Test 5</a></div>
          </div>
        </div>
      </li>
      <li class=\\"mt-4\\"><a href=\\"https://seinet.sei.cmu.edu\\" class=\\"link link-primary link-cta\\">
          View All
        </a></li>
    </ul>
  </div>
</div>"
`;

exports[`TopFiveChart.vue matches snapshot with urls, doNotLinkEntries 1`] = `
"<div data-id=\\"sds-top-five-chart\\" class=\\"space-y-4 chart\\">
  <h3 class=\\"text-2xl font-bold\\">
    Top Five Chart (with urls, doNotLinkEntries)
  </h3>
  <div>
    <ul class=\\"space-y-6\\">
      <li>
        <div class=\\"flex\\">
          <div class=\\"flex-grow\\">
            <div class=\\"mb-2\\">
              <div role=\\"progressbar\\" title=\\"40\\" aria-valuenow=\\"40\\" aria-valuemin=\\"0\\" aria-valuemax=\\"40\\" class=\\"h-6 mr-2 rounded bg-blue-900\\" style=\\"width: 100%;\\"><span class=\\"sr-only\\">40
                  Test 1</span></div>
            </div>
            <div class=\\"text-sm font-semibold chart-label-section\\">
              40
              ·
              <span>Test 1</span></div>
          </div>
        </div>
      </li>
      <li>
        <div class=\\"flex\\">
          <div class=\\"flex-grow\\">
            <div class=\\"mb-2\\">
              <div role=\\"progressbar\\" title=\\"30\\" aria-valuenow=\\"30\\" aria-valuemin=\\"0\\" aria-valuemax=\\"40\\" class=\\"h-6 mr-2 rounded bg-blue-700\\" style=\\"width: 75%;\\"><span class=\\"sr-only\\">30
                  Test 2</span></div>
            </div>
            <div class=\\"text-sm font-semibold chart-label-section\\">
              30
              ·
              <span>Test 2</span></div>
          </div>
        </div>
      </li>
      <li>
        <div class=\\"flex\\">
          <div class=\\"flex-grow\\">
            <div class=\\"mb-2\\">
              <div role=\\"progressbar\\" title=\\"15\\" aria-valuenow=\\"15\\" aria-valuemin=\\"0\\" aria-valuemax=\\"40\\" class=\\"h-6 mr-2 rounded bg-blue-500\\" style=\\"width: 37.5%;\\"><span class=\\"sr-only\\">15
                  Test 3</span></div>
            </div>
            <div class=\\"text-sm font-semibold chart-label-section\\">
              15
              ·
              <span>Test 3</span></div>
          </div>
        </div>
      </li>
      <li>
        <div class=\\"flex\\">
          <div class=\\"flex-grow\\">
            <div class=\\"mb-2\\">
              <div role=\\"progressbar\\" title=\\"10\\" aria-valuenow=\\"10\\" aria-valuemin=\\"0\\" aria-valuemax=\\"40\\" class=\\"h-6 mr-2 rounded bg-blue-300\\" style=\\"width: 25%;\\"><span class=\\"sr-only\\">10
                  Test 4</span></div>
            </div>
            <div class=\\"text-sm font-semibold chart-label-section\\">
              10
              ·
              <span>Test 4</span></div>
          </div>
        </div>
      </li>
      <li>
        <div class=\\"flex\\">
          <div class=\\"flex-grow\\">
            <div class=\\"mb-2\\">
              <div role=\\"progressbar\\" title=\\"3\\" aria-valuenow=\\"3\\" aria-valuemin=\\"0\\" aria-valuemax=\\"40\\" class=\\"h-6 mr-2 rounded bg-blue-100\\" style=\\"width: 7.5%;\\"><span class=\\"sr-only\\">3
                  Test 5</span></div>
            </div>
            <div class=\\"text-sm font-semibold chart-label-section\\">
              3
              ·
              <span>Test 5</span></div>
          </div>
        </div>
      </li>
      <!---->
    </ul>
  </div>
</div>"
`;
